### PR TITLE
246 broken data compatibility

### PR DESCRIPTION
**Pull request by pyakushin**

### CODE REVIEW

**Comment by Justarone** [comment]
I finally found rpm package, on which broken compatibility was found and... it still exists. As I remember we have discussed 2 solutions:
1. Revert library version
2. Move code from library to get rid of this dangerous dependency and control it ourselves

I see, that you have fixed some parameters to prevent this kind of problems in the future (and in this case code looks good to me (maybe we also should fix crc version "=2.0.0")), but it doesn't solve existing problem. @ikopylov , need your advice.

**Comment by Justarone** [comment]
And I don't think that it's a good idea to update tokio version here either, because if it will break something here, we will revert breaking changes, but we'll revert fixing changes (for this problem) also

**Comment by ikopylov** [comment]
We decided to break backward compatibility and freeze the library version at the newest one. The reason is that the next major release should also contain BPlusTree index (https://github.com/qoollo/pearl/pull/93), which is also a breaking change, so we allow it here as well. 

@pyakushin But it is still important to increment index file version.

> maybe we also should fix crc version "=2.0.0"

CRC is a super common algorithm, so the calculation can only change due to the bug in the library. However, it would be nice to write a test, that checks CRC calculation.

File: src/lib.rs [hunk 1]
```diff
@@ -70,14 +70,16 @@ pub use rio;
 pub use storage::{Builder, Key, Storage};
 
 mod prelude {
+    pub const CASTAGNOLI: Crc<u32> = Crc::<u32>::new(&CRC_32_ISCSI);
```
Comment (idruzhitskiy) [inline src/lib.rs:73]: IMO it's better to name it CRC32C

File: tests/tests.rs [hunk 2]
```diff
@@ -19,6 +20,17 @@ mod common;
 
 use common::KeyTest;
 
+#[test]
+fn test_hash() {
+    let data = (0..10).collect::<Vec<u8>>();
```
Comment (idruzhitskiy) [inline tests/tests.rs:25]: Maybe there should be multiple cases, 3-5, including 0 and something large

File: src/lib.rs (hunk 1 above)
Comment (pyakushin) [reply src/lib.rs:73]: Ok

**Comment by pyakushin** [comment]
> And I don't think that it's a good idea to update tokio version here either, because if it will break something here, we will revert breaking changes, but we'll revert fixing changes (for this problem) also

1. I double checked, there are no breaking changes. 
2. But fixed a couple of memory leaks

**Review by Justarone — APPROVED** [review]

**Pull request merged**